From 124549bb53f3071dfc594b7e2aec96c4e0bd3ea2 Mon Sep 17 00:00:00 2001
From: "Samuel A. Winchenbach" <swinchen@gmail.com>
Date: Tue, 26 May 2020 15:52:18 +0530
Subject: [PATCH 1367/1852] spi-xilinx.c: correct multibyte writes

commit 31991b3386eebec17581e8752fc584cbad9fd1c5 from
https://github.com/Xilinx/linux-xlnx.git

When bits-per-word is greater than 8 only the first byte is sent; this
is caused by an incorrect cast in the macros used to fill the tx fifo.

Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Signed-off-by: Samuel A. Winchenbach <swinchen@gmail.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/spi/spi-xilinx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-xilinx.c b/drivers/spi/spi-xilinx.c
index d23632784c15..ed1005c00ddd 100644
--- a/drivers/spi/spi-xilinx.c
+++ b/drivers/spi/spi-xilinx.c
@@ -178,7 +178,7 @@ static void xspi_fill_tx_fifo_##size(struct xilinx_spi *xqspi)		\
 	u32 data = 0;							\
 	for (i = 0; i < count; i += (size / 8)) {			\
 		if (xqspi->tx_ptr)					\
-			data = (type)((u8 *)xqspi->tx_ptr)[i];		\
+			data = *(type *)&xqspi->tx_ptr[i];		\
 		writel_relaxed(data, (xqspi->regs + XSPI_TXD_OFFSET));	\
 	}								\
 	xqspi->bytes_to_transfer -= count;				\
-- 
2.31.1

